lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -20,18 +20,46 @@ import ValidationMessage from './ValidationMessage';
 export default function SubmitGradeForm({ disabled = false, student }) {
   const [showError, setShowError] = useState(false); // Is there a validation error message to show?
   const [gradeErrorMessage, setGradeErrorMessage] = useState(''); // The actual validation error message.
+  const [grade, setGrade] = useState('');
 
-  const [networkError, setNetworkError] = useState(''); // if there is an error when submitting grade
-  const [requestStatus, setRequestStatus] = useState(''); // ajax request state, one of ('', 'fetching', 'error', 'done')
+  const [networkError, setNetworkError] = useState(''); // If there is an error when submitting grade
+  const [requestStatus, setRequestStatus] = useState(''); // Ajax request state, one of ('', 'fetching', 'error', 'done')
+  const [fetchStatus, setFetchStatus] = useState(''); // Ajax request state, one of ('', 'fetching', 'error', 'done')
 
   const { authToken } = useContext(Config);
 
+  // Used to handle keyboard input effects.
+  const refInput = useRef(null);
+
   useEffect(() => {
     setRequestStatus(''); // clear out network status
-  }, [student]);
 
-  // Used to handle keyboard input effects.
-  const refInput = useRef(null);
+    // Fetch the grade from the service api
+    let fetchedGradeDone = false; // flag to prevent updating more than once
+    if (Object.entries(student).length) {
+      // See https://github.com/facebook/react/issues/14326 for async in useEffect
+      (async () => {
+        try {
+          setFetchStatus('fetching');
+          setGrade(''); // Clear the older grade so we don't show the wrong grade with the new student
```

Comment:
Is there ever a state in which you wouldn't initially `setGrade('')` in this `useEffect` callback? Right now, you've got an `else` branch that exists solely to `setGrade('')` but I'm wondering if it really happens for all branches (making that `else` potentially unnecessary?).

Reply:
I think you may be correct and I wrote the else condition before I wrote line 44.